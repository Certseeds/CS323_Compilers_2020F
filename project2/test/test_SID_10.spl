struct wow1
{
    int a1;
    float array1[1];
};
struct wow2
{
    int a2;
    float array2[2];
}a3;
struct wow3
{
    int a4;
    float array3[3];
}a5;
struct wow4
{
    int a6;
    float array4[4];
}a7[1];
struct wow5
{
    int a8;
    float array5[4];
}a9,a10[1];
int function(){
    return 0;
}
float function(){
    return 1.0;
}
int mainFunction(){
    struct wow1 wow1Def;
    struct wow2 wow2Def;
    struct wow3 wow3Def;
    struct wow4 wow4Def;
    struct wow5 wow5Def;
    struct wow5 wow5Def;
    wow6;
    otherfunction();
    wow3 = wow4;
    function() = 1;
    wow3 = wow4 + wow5;
    function(1,2,3,4,5);
    a10[0].array5[2][2] =  1;
    a7[0] = a8();
    a10[a7[0]] = 2;
    return wow1;
}