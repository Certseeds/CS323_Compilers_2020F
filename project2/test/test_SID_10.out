Error type 4 at Line 29: redefine function: function
Error type 3 at Line 38: redefine variable: wow5Def
Error type 1 at Line 39: undefined variable: wow6
Error type 2 at Line 40: undefined function: otherfunction
Error type 5 at Line 41: unmatching types on both sides of assignment
Error type 6 at Line 42: left side in assignment is rvalue
Error type 5 at Line 42: unmatching types on both sides of assignment
Error type 7 at Line 43: binary operation on non-number variables
Error type 7 at Line 43: binary operation on non-number variables
Error type 5 at Line 43: unmatching types on both sides of assignment
Error type 9 at Line 44: invalid argument number for function, expect 0, got 5
Error type 10 at Line 45: indexing on non-array variable
Error type 5 at Line 45: unmatching types on both sides of assignment
Error type 2 at Line 46: undefined function: a8
Error type 5 at Line 46: unmatching types on both sides of assignment
Error type 12 at Line 47: indexing by non-integer
Error type 5 at Line 47: unmatching types on both sides of assignment
Error type 8 at Line 48: incompatiable return type
